src/solver/opamCudf.ml

```diff
@@ -951,7 +938,42 @@ let extract_explanations packages cudfnv2opam unav_reasons reasons =
           explanations, ct_chains)
       ([], ct_chains) reasons
   in
-  List.rev explanations
+
+  let explanations =
+    match explanations with
+    | `Missing (_, sdeps, fdeps) :: rest
+      when  List.for_all (function `Missing (_, sdeps', fdeps') -> sdeps = sdeps' && fdeps = fdeps' | _ -> false) rest ->
+        [`Missing (None, sdeps, fdeps)]
+    | _ -> explanations
+
+  in
```

Comment:
```suggestion
  let explanations =
    let same_depexts sdeps fdeps =
      List.for_all (function
          | `Missing (_, sdeps', fdeps') -> sdeps = sdeps' && fdeps = fdeps'
          | _ -> false)
    in
    match explanations with
    | `Missing (_, sdeps, fdeps) :: rest when same_depexts sdeps fdeps rest ->
      [`Missing (None, sdeps, fdeps)]
    | _ -> explanations
  in
```
Nitpicking on the style/line length ;) feel free to drop

Reply:
It's much neater that way :)